src/pages/iou/request/step/IOURequestStepConfirmation.js

```diff
@@ -265,7 +268,26 @@ function IOURequestStepConfirmation({
             }
 
             if (receiptFile) {
-                requestMoney(selectedParticipants, trimmedComment, receiptFile);
+                getCurrentPosition(
+                    (successData) => {
+                        requestMoney(selectedParticipants, trimmedComment, receiptFile, {
+                            lat: successData.coords.latitude,
+                            long: successData.coords.longitude,
+                        });
+                    },
+                    (errorData) => {
+                        Log.info('[IOURequestStepConfirmation] getCurrentPosition failed', false, errorData);
+                        // When there is an error, the money can still be requested, it just won't include the GPS coordinates
+                        requestMoney(selectedParticipants, trimmedComment, receiptFile);
+                    },
+                    {
+                        // It's OK to get a cached location that is up to an hour old because the only accuracy needed is the country the user is in
+                        maximumAge: 1000 * 60 * 60,
+
+                        // 15 seconds, don't way too long because the server can always fall back to using the IP address
```

Comment:
NAB:
```suggestion
                        // 15 seconds, don't wait too long because the server can always fall back to using the IP address
```

Reply:
What's the usual time to get this data, 15 seconds sounds long... is the app blocked and the user waiting for this or does it happen in the background?

Reply:
On the distance request, this is set to 30 seconds. Every time I've tested it, it only takes milliseconds, so I think it's a very extreme timeout case. It would make sense that it does block the app.